src/Uno.UI/Microsoft/UI/Xaml/Controls/Common/ColorConversion.cs

```diff
@@ -0,0 +1,346 @@
+			double intermediateColorPercentage = hue / 60 - sextant;
+			double max = chroma + min;
+
+			double r = 0;
+			double g = 0;
+			double b = 0;
+
+			switch (sextant)
+			{
+				case 0:
+					r = max;
+					g = min + chroma * intermediateColorPercentage;
+					b = min;
+					break;
+				case 1:
+					r = min + chroma * (1 - intermediateColorPercentage);
+					g = max;
+					b = min;
+					break;
+				case 2:
+					r = min;
+					g = max;
+					b = min + chroma * intermediateColorPercentage;
+					break;
+				case 3:
+					r = min;
+					g = min + chroma * (1 - intermediateColorPercentage);
+					b = max;
+					break;
+				case 4:
+					r = min + chroma * intermediateColorPercentage;
+					g = min;
+					b = max;
+					break;
+				case 5:
+					r = max;
+					g = min;
+					b = min + chroma * (1 - intermediateColorPercentage);
+					break;
+			}
+
+			return new Rgb(r, g, b);
+		}
+
+		public static Rgb HexToRgb(string input)
+		{
+			var (rgb, a) = HexToRgba(input);
+			return rgb;
+		}
+
+		public static string RgbToHex(Rgb rgb)
+		{
+			byte rByte = (byte)Math.Round(rgb.R * 255.0);
+			byte gByte = (byte)Math.Round(rgb.G * 255.0);
+			byte bByte = (byte)Math.Round(rgb.B * 255.0);
+
+			UInt64 hexValue = ((UInt64)rByte << 16) + ((UInt64)gByte << 8) + (UInt64)bByte;
+
+			// Uno Doc: Using C#/.net string methods instead
+			string hexString = string.Format("#{0:X6}", hexValue);
+
+			// We'll size this string to accommodate "#XXXXXX" - i.e., a full RGB number with a # sign.
+			//wchar_t hexString[8];
+			//winrt::check_hresult(StringCchPrintfW(&hexString[0], 8, L"#%06X", hexValue));
+
+			return hexString;
+		}
+
+		public static (Rgb, double) HexToRgba(string input)
+		{
+			// The input always begins with a #, so we'll move past that.
+			input = input?.Length > 1 ? input?.Substring(1) : input;
+
+			var hexValue = TryParseInt(input, 16);
+
+			// If we failed to parse the string into an integer, then we'll return all -1's.
+			// ARGB values can never be negative, so this is a convenient error state to use
+			// to indicate that this value should not actually be used.
+			if (!hexValue.HasValue)
+			{
+				return (new Rgb(-1, -1, -1), -1);
+			}
+
+			var hex = hexValue.Value;
+			byte a = (byte)((hex & 0xff000000) >> 24);
+			byte r = (byte)((hex & 0x00ff0000) >> 16);
+			byte g = (byte)((hex & 0x0000ff00) >> 8);
+			byte b = (byte)(hex & 0x000000ff);
+
+			return (new Rgb(r / 255.0, g / 255.0, b / 255.0), a / 255.0);
+		}
+
+		public static string RgbaToHex(Rgb rgb, double alpha)
+		{
+			byte aByte = (byte)Math.Round(alpha * 255.0);
+			byte rByte = (byte)Math.Round(rgb.R * 255.0);
+			byte gByte = (byte)Math.Round(rgb.G * 255.0);
+			byte bByte = (byte)Math.Round(rgb.B * 255.0);
+
+			UInt64 hexValue = ((UInt64)aByte << 24) + ((UInt64)rByte << 16) + ((UInt64)gByte << 8) + ((UInt64)bByte & 0xff);
+
+			// Uno Doc: Using C#/.net string methods instead
+			string hexString = string.Format("#{0:X8}", hexValue);
```

Comment:
Codacy found an issue: [Use the overload that takes a 'CultureInfo' or 'IFormatProvider' parameter.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6121259)